process/economics/globalSettings.go

```diff
@@ -0,0 +1,142 @@
+package economics
+
+import (
+	"github.com/multiversx/mx-chain-core-go/core/check"
+	"math"
+	"sort"
+	"sync"
+
+	"github.com/multiversx/mx-chain-go/config"
+	"github.com/multiversx/mx-chain-go/process"
+)
+
+type inflationForEpochCompound struct {
+	enableEpoch  uint32
+	maxInflation float64
+}
+
+type globalSettingsHandler struct {
+	minInflation                 float64
+	yearSettings                 map[uint32]*config.YearSetting
+	chainParameters              process.ChainParametersHandler
+	tailInflationActivationEpoch uint32
+	startYearInflation           float64
+	inflationForEpoch            []inflationForEpochCompound
+	decayPercentage              float64
+	mutYearSettings              sync.RWMutex
+}
+
+var numMillisecondsPerSeconds = uint64(1000)
+var numSecondsPerMinute = uint64(60)
+var numMinutesPerHour = uint64(60)
+var numHoursPerDay = uint64(24)
+var numDaysInYear = uint64(365)
+var numberOfMillisecondsInYear = numMillisecondsPerSeconds * numSecondsPerMinute * numMinutesPerHour * numHoursPerDay * numDaysInYear
+
+// newGlobalSettingsHandler creates a new global settings provider
+func newGlobalSettingsHandler(
+	economics *config.EconomicsConfig,
+	chainParameters process.ChainParametersHandler,
+) (*globalSettingsHandler, error) {
+	if check.IfNil(chainParameters) {
+		return nil, process.ErrNilChainParametersHandler
+	}
+
+	g := &globalSettingsHandler{
+		minInflation:                 economics.GlobalSettings.MinimumInflation,
+		yearSettings:                 make(map[uint32]*config.YearSetting),
+		tailInflationActivationEpoch: economics.GlobalSettings.TailInflation.EnableEpoch,
+		startYearInflation:           economics.GlobalSettings.TailInflation.StartYearInflation,
+		decayPercentage:              economics.GlobalSettings.TailInflation.DecayPercentage,
+		mutYearSettings:              sync.RWMutex{},
+		chainParameters:              chainParameters,
+	}
+
+	g.yearSettings = make(map[uint32]*config.YearSetting)
+	for _, yearSetting := range economics.GlobalSettings.YearSettings {
+		g.yearSettings[yearSetting.Year] = &config.YearSetting{
+			Year:             yearSetting.Year,
+			MaximumInflation: yearSetting.MaximumInflation,
+		}
+	}
+
+	err := g.calculateInflationForEpochCompound()
+	if err != nil {
+		return nil, err
+	}
+
+	if isPercentageInvalid(g.minInflation) ||
+		isPercentageInvalid(g.startYearInflation) ||
+		isPercentageInvalid(g.decayPercentage) {
+		return nil, process.ErrInvalidInflationPercentages
+	}
+
+	return g, nil
+}
+
+func (g *globalSettingsHandler) calculateInflationForEpochCompound() error {
```

Comment:
we will need some unit tests on this, as it is changed now with chain parameters